patches/chromium/feat_add_frame-level_find_request_managers_in_web_contents_impl.patch

```diff
@@ -0,0 +1,419 @@
+   if (search_text.empty()) {
+@@ -4973,13 +4975,13 @@ void WebContentsImpl::Find(int request_id,
+     return;
+   }
+ 
+-  GetOrCreateFindRequestManager()->Find(request_id, search_text,
++  GetOrCreateFindRequestManager(rfh)->Find(request_id, search_text,
+                                         std::move(options));
+ }
+ 
+-void WebContentsImpl::StopFinding(StopFindAction action) {
++void WebContentsImpl::StopFinding(StopFindAction action, RenderFrameHost* rfh) {
+   OPTIONAL_TRACE_EVENT0("content", "WebContentsImpl::StopFinding");
+-  if (FindRequestManager* manager = GetFindRequestManager())
++  if (FindRequestManager* manager = GetFindRequestManager(rfh))
+     manager->StopFinding(action);
+ }
+ 
+@@ -7934,7 +7936,12 @@ std::unique_ptr<WebUIImpl> WebContentsImpl::CreateWebUI(
+   return nullptr;
+ }
+ 
+-FindRequestManager* WebContentsImpl::GetFindRequestManager() {
++FindRequestManager* WebContentsImpl::GetFindRequestManager(RenderFrameHost* root_frame) {
++  if (root_frame) {
++    auto it = frame_find_request_managers_.find(root_frame);
++    return it == frame_find_request_managers_.end() ? nullptr : it->second.get();
++  }
++
+   for (auto* contents = this; contents;
+        contents = contents->GetOuterWebContents()) {
+     if (contents->find_request_manager_)
+@@ -7944,14 +7951,20 @@ FindRequestManager* WebContentsImpl::GetFindRequestManager() {
+   return nullptr;
+ }
+ 
+-FindRequestManager* WebContentsImpl::GetOrCreateFindRequestManager() {
+-  if (FindRequestManager* manager = GetFindRequestManager())
++FindRequestManager* WebContentsImpl::GetOrCreateFindRequestManager(RenderFrameHost *root_frame) {
++  if (FindRequestManager* manager = GetFindRequestManager(root_frame))
+     return manager;
+ 
+   DCHECK(!browser_plugin_guest_ || GetOuterWebContents());
+ 
+-  // No existing FindRequestManager found, so one must be created.
+-  find_request_manager_ = std::make_unique<FindRequestManager>(this);
++  if (root_frame) {
++    // No existing FindRequestManager found, so one must be created.
++    frame_find_request_managers_[root_frame] = std::make_unique<FindRequestManager>(this, root_frame);
++  }
++  else {
++    // No existing FindRequestManager found, so one must be created.
++    find_request_manager_ = std::make_unique<FindRequestManager>(this);
++  }
+ 
+   // Concurrent find sessions must not overlap, so destroy any existing
+   // FindRequestManagers in any inner WebContentses.
+@@ -7964,20 +7977,38 @@ FindRequestManager* WebContentsImpl::GetOrCreateFindRequestManager() {
+           STOP_FIND_ACTION_CLEAR_SELECTION);
+       web_contents_impl->find_request_manager_.release();
+     }
++    for (auto& it : web_contents_impl->frame_find_request_managers_) {
++      it.second->StopFinding(
++          STOP_FIND_ACTION_CLEAR_SELECTION);
++    }
++    web_contents_impl->frame_find_request_managers_.clear();
++  }
++  for (RenderFrameHost* frame : root_frame->GetFramesInSubtree()) {
```

Comment:
what is the loop for ? the previous iteration over `WebContents` also seem to clear any `frame_find_request_managers_` associated with it.

Reply:
The previous iteration goes over all the WebContents within `this`, but excludes `this` (so `web_contents_impl != this` when `clear` is called). This iteration goes over all the subframes under `root_frame` that are within the WebContents `this`.